x-pack/plugins/fleet/server/saved_objects/index.ts

```diff
@@ -540,6 +541,16 @@ export const getSavedObjectTypes = (
             },
           ],
         },
+        '10': {
+          changes: [
+            {
+              type: 'mappings_addition',
+              addedMappings: {
+                package: { properties: { requires_root: { type: 'boolean' } } },
```

Comment:
I'm not sure if we can backfill the  `requires_root` flag for existing package policies, as it would require async calls to get the package info. 
Alternatively we could have a hardcoded list of root packages in the migration function.

Reply:
How can a mapping be optional?

I added the population to the create/update API, there will be some packages that require root (not all), and the package-policy field is used to show a warning on the UI. Ideally we should show the warning on existing package policies too. Filling in on read could work too.

Reply:
Added a data backfill with hardcoded list of packages in 8.15, I think this is the simplest and least risky change.

Reply:
@jloleysens are you ok to approve with the latest changes?